timemachine/cpp/src/potential.cu

```diff
@@ -122,4 +119,36 @@ void Potential::execute_host_du_dx(
     gpuErrchk(cudaFree(d_box));
 };
 
+void Potential::fixed_to_float(
+    const int N,
+    const int P,
+    const unsigned long long *du_dx,
+    const double *du_dp,
+    const unsigned long long *du_dl,
+    const unsigned long long *u,
+    double *du_dx_out,
+    double *du_dp_out,
+    double *du_dl_sum,
+    double *u_sum) {
+
+    for (int i = 0; i < N * D; i++) {
+        du_dx_out[i] = FIXED_TO_FLOAT<double>(du_dx[i]);
+    }
+
+    for (int i = 0; i < P; i++) {
+        du_dp_out[i] = du_dp[i]; // TODO: just a pass-thru for now; update when du_dp is fixed-point
```

Comment:
In #556 , the subclass implementations will add the appropriate conversion here

Reply:
Makes sense! It's probably actually good to be explicit about the conversion for each type of potential, since each functional form will have different sized (and different ranges) of du_dps.

Reply:
Though I think du_dx and du_dls may not need too much specialization, since on average the forces are fairly well behaved across the potentials (for MD, we definitely have an upper bound, otherwise the integrator will fall apart). du_dps and maybe u are more finnicky and variable across different classes.

Reply:
> Though I think du_dx and du_dls may not need too much specialization

Makes sense. Maybe can pull out the conversion for parameter derivatives into a separate method so that subclasses can override just that.

Reply:
Resolving this for now; I'll consider splitting out the du_dp conversion into a separate method when it comes time to override this in #556 .